app/queries/keyword_search_entries_query.rb

```diff
@@ -0,0 +1,41 @@
+# frozen_string_literal: true
+
+class KeywordSearchEntriesQuery
+  def initialize(relation = KeywordSearchEntry)
+    @relation = relation
+  end
+
+  def with_keyword(keyword_id)
```

Comment:
How about

```suggestion
  def by_keyword_id(keyword_id)
```

Reply:
My two cents: the following are all equivalent to me 😅 

```suggestion
  def with_keyword(keyword_id)
```

```suggestion
  def by_keyword(keyword_id)
```

```suggestion
  def for_keyword(keyword_id)
```

The key is to be consistent throughout the codebase 💡